kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -98,6 +109,44 @@ public static Map<String, Map<String, Variant<?>>> buildSettings(NetworkProperti
         return newConnectionSettings;
     }
 
+    private static Map<String, Variant<?>> build80211WirelessSecurity8021xSettings(NetworkProperties props,
+            String deviceId) {
+
+        String propMode = props.get(String.class, KURA_PROPS_KEY_WIFI_MODE, deviceId);
+
+        String eap = props.get(String.class, "net.interface.%s.config.802-1x.eap", deviceId, propMode.toLowerCase());
+        String phase2 = props.get(String.class, "net.interface.%s.config.802-1x.innerAuth", deviceId,
+                propMode.toLowerCase());
+
+        Map<String, Variant<?>> settings = new HashMap<>();
+
+        NM8021xEapAndPhase2Configurator eapConf;
+        NM8021xEapAndPhase2Configurator phase2Conf;
+
+        // Configure Eap Method
```

Comment:
better names need to be picked for this.

The code that performs configuration is encapsulated by its respective class.

When we want to add another security protocol, we create a new class for it, put the respective config in the class and finally link it up in the switch structure

Reply:
https://github.com/eclipse/kura/blob/735af3fdcef212adcbff14394bb7d8e16b6ac555/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java#L123-L145

Reply:
I don't really like where this is going tbh 🤔  The purpose of the `NMSettingsConverter` class is to have a place where we can have a direct mapping between the Kura properties/settings and the NM ones. Having the NM properties hidden in the `NM8021xEapAndPhase2Configurator` is somewhat fighting against this (even though it's cleaner from the code point of view) and is not following the already established convention... Keep in mind that, at the end of the day, we're just translating a map into another map. I prefer a simpler approach... but I might be lacking context ATM

We can discuss this next time you're available. Don't worry for now :)